### ISSUE TITLE
OLED Screen not working

### ISSUE THREAD

**Issue by hoofta** [HackerHouseYT/Smart-Bartender#3]
I have yet to get my OLED screen to work. I have tried two different OLEDs; Adafruit and a generic brand. Admittedly, both screens have cracked at the bottom left while installing them but where they cracked seem to be inconsequential. Since I've gone through two of them I can't help but think that something else is wrong.

**Comment by hoofta** [comment]
I noticed on the back of the Adafruit OLED screen there is a "For I2C Close SJ1/2". The below it there are 2 small metal contacts above SJ1, below that 2 small metal contacts above SJ2. Since I2C is referenced during the OLED setup, should something be done with these contacts?

**Comment by hoofta** [comment]
I have checked, rechecked, and even just unplugged all wires from GPIO pins, and reinstalled Raspbian. I have literally started from scratch and the OLED screen is still not working.
(╯°□°）╯︵ ┻━┻

**Comment by hoofta** [comment]
So no one else is having OLED issues? I just saw this posted 2 days ago from The Raspberry Pi Guy. The code is 5 years old and apparently doesn't work with pi 3/3B+

https://github.com/the-raspberry-pi-guy/OLED/issues/9

**Comment by olliegg123** [comment]
I'm using an I2C OLED screen and I've managed to get it to work, if you need any help.

**Comment by hoofta** [comment]
Thanks for the offer Olliegg123. Was that you that just posted on the hackster page with the fix for the OLED and the double drinks?

**Comment by olliegg123** [comment]
Yeah! It’s been me and my friend doing it, he’s posted a link to his code on the first issue created on here.

**Comment by hoofta** [comment]
My hat is off to both of you. This will be the first thing I do when I get home tonight! As of Friday night, I thought I had exhausted all my possibilities. Thanks for starting my Monday off right :)

**Comment by olliegg123** [comment]
Not a problem! If you need any help, just ask us. We’ll try our best. We did 20 tests regarding the double drinks, 10 physical button presses and 10 over SSH. No double drinks after our fixes. Only issue is our code doesn’t use the LEDs and we made it smoother by taking out threading completely. But it seems to be working perfectly!

**Comment by hoofta** [comment]
VERY COOL! I did include the LEDs in my build but instead of putting them up top, I replaced the bottom coaster with an infinity mirror/kaleidoscope. So when the drink is being made it should look like it's hovering over nothing. When I get home and plug your code in, I'll see if that did anything for the LEDs on my end. Thanks again!

**Comment by olliegg123** [comment]
Awesome idea! Keep us updated!

**Comment by bjensen** [comment]
@olliegg123 @danzibob thanks for the effort. When I try to use the danzibob fork I get an error of missing the module lib_oled96. I installed pillow and luma-oled using https://luma-oled.readthedocs.io/en/latest/install.html but I still get the error.

**Comment by bjensen** [comment]
@hoofta are you still struggling with getting the OLED to work? I am too :-/

**Comment by Joe2824** [comment]
Today I receive my OLED SSD1306 but it doesn't work at all :(

**Comment by joe2824** [comment]
I change the code to the Adafruit SSD1306 Library feel free to use.

https://github.com/Joe2824/Smart-Bartender
